network-libp2p/src/discovery/peer_contacts.rs

```diff
@@ -97,8 +111,12 @@ impl PeerContact {
     }
 
     /// Adds a set of addresses
-    pub fn add_addresses(&mut self, addresses: Vec<Multiaddr>) {
-        self.addresses.extend(addresses)
+    pub fn add_addresses(&mut self, addresses: Vec<Multiaddr>) -> Result<(), PeerContactError> {
+        if addresses.len() + self.addresses.len() > Self::MAX_ADDRESSES {
+            return Err(PeerContactError::AdvertisedAddressesExceeded);
+        }
+        self.addresses.extend(addresses);
+        Ok(())
```

Comment:
Maybe this could add addresses until it overflows, so that at least some are recorded?